src/components/Dependencies/Dependencies.tsx

```diff
@@ -1,117 +1,78 @@
+import { PackageDependencies } from '../../../types/packageMeta';
+import { DetailContext } from '../../pages/Version';
 import NoItems from '../NoItems';
 
 import { CardWrap, Heading, Tags, Tag } from './styles';
 
-type DepDetailProps = {
-  name: string;
-  version: string;
-  onLoading?: () => void;
-} & RouteComponentProps;
-
-interface DepDetailState {
-  name: string;
-  version: string;
+interface DependencyBlockProps {
+  title: string;
+  dependencies: PackageDependencies;
 }
 
-class DepDetail extends Component<DepDetailProps, DepDetailState> {
-  constructor(props: DepDetailProps) {
-    super(props);
-    const { name, version } = this.props;
+const DependencyBlock: React.FC<DependencyBlockProps> = ({ title, dependencies }) => {
+  const { enableLoading } = useContext(DetailContext);
+  const history = useHistory();
 
-    this.state = {
-      name,
-      version,
-    };
-  }
+  const deps = Object.entries(dependencies);
 
-  public render(): ReactElement<HTMLElement> {
-    const { name, version } = this.state;
-    const tagText = `${name}@${version}`;
-    return <Tag className={'dep-tag'} clickable={true} label={tagText} onClick={this.handleOnClick} />;
-  }
+  function handleClick(name: string): void {
+    enableLoading && enableLoading();
 
-  private handleOnClick = () => {
-    const { name } = this.state;
-    const { onLoading, history } = this.props;
-
-    onLoading && onLoading();
     history.push(`/-/web/detail/${name}`);
-  };
-}
+  }
 
-const WrapperDependencyDetail = withRouter(DepDetail);
+  return (
+    <CardWrap>
+      <CardContent>
+        <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
+        <Tags>
+          {deps.map(([name, version]) => (
+            // eslint-disable-next-line
+            <Tag className={'dep-tag'} clickable={true} key={name} label={`${name}@${version}`} onClick={() => handleClick(name)} />
+          ))}
+        </Tags>
+      </CardContent>
+    </CardWrap>
+  );
+};
+
+function hasKeys(object?: { [key: string]: any }): boolean {
+  return !!object && Object.keys(object).length > 0;
+}
 
-class DependencyBlock extends Component<{ title: string; dependencies: [] }> {
-  public render(): ReactElement<HTMLElement> {
-    const { dependencies, title } = this.props;
-    const deps = Object.entries(dependencies) as [];
+const Dependencies: React.FC<{}> = () => {
+  const { packageMeta } = useContext(DetailContext);
 
-    return (
-      <DetailContextConsumer>
-        {({ enableLoading }) => {
-          return (
-            <CardWrap>
-              <CardContent>
-                <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
-                <Tags>{this.renderTags(deps, enableLoading)}</Tags>
-              </CardContent>
-            </CardWrap>
-          );
-        }}
-      </DetailContextConsumer>
-    );
+  if (!packageMeta) {
+    throw new Error('DetailContext is required');
```

Comment:
I believe the error message should be : `packageMeta is required`

Reply:
`packageMeta` is only a field of `context` so the message is actually correct :)

Reply:
What's about `packageMeta is required at DetailContext`? All win.